### PR TITLE
KOGITO-3364: [DMN Designer] Multiple DRDs support - When a node is deleted in the DRG, undo/redo operations do not work properly

### PR DESCRIPTION
**Pull request by danielzhe**
**JIRA**: [KOGITO-3364](https://issues.redhat.com/browse/KOGITO-3364)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Comment by danielzhe** [comment]
Working on SonarCloud...

**Comment by danielzhe** [comment]
jenkins do fdb

**Comment by karreiro** [comment]
Jenkins do fdb.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/session/DMNCanvasHandler.java [hunk 1]
```diff
@@ -121,5 +121,7 @@ void superAddChild(final Element parent,
                        final Element child) {
         super.addChild(parent, child);
     }
+
+
```
Comment (karreiro) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/session/DMNCanvasHandler.java:125]: Can we remove these extra lines? :-)

**Review by karreiro — APPROVED** [review]
Great stuff, @danielzhe LGTM and works fine!

I've left just a non-blocking comment ;-)

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/session/DMNCanvasHandler.java (hunk 1 above)
Comment (danielzhe) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/session/DMNCanvasHandler.java]: Removed! Thank you, @karreiro !

**Review by karreiro — APPROVED** [review]

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/factory/graph/DMNSetConnectionSourceNodeCommand.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                                             final Edge<? extends View, Node> edge,
+                                             final Connection connection,
+                                             final GraphsProvider graphsProvider) {
+        super(sourceNode, edge, connection);
+        this.graphsProvider = graphsProvider;
+        this.diagramId = ContentDefinitionIdUtils.getDiagramId(edge);
+    }
+
+    @Override
+    public Node<? extends View<?>, Edge> getSourceNode(final GraphCommandExecutionContext context) {
```
Comment (jstastny-cz) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/factory/graph/DMNSetConnectionSourceNodeCommand.java:51]: sonar reporting critical code smells when having generic wildcards in return types: https://sonarcloud.io/organizations/kiegroup/rules?open=java%3AS1452&rule_key=java%3AS1452

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/factory/graph/DMNSetConnectionSourceNodeCommand.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        return super.getNode(context, uuid);
+    }
+
+    boolean commandBelongsToAnotherGraph() {
+        return getDiagramId().isPresent()
+                && !isTheCurrentDiagram(getDiagramId(), graphsProvider);
+    }
+
+    Graph getEdgesGraph() {
+        if (getDiagramId().isPresent()) {
```
Comment (jstastny-cz) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/factory/graph/DMNSetConnectionSourceNodeCommand.java:100]: here sonar reports bug on not calling isPresent() before get() on optional ... and the reason IMO is, that it's not called on a variable (method returns new optional each time). Please introduce a variable.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/factory/graph/DMNSetConnectionTargetNodeCommand.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        return super.getNode(context, uuid);
+    }
+
+    boolean commandBelongsToAnotherGraph() {
+        return getDiagramId().isPresent()
+                && !isTheCurrentDiagram(getDiagramId(), graphsProvider);
+    }
+
+    Graph getEdgesGraph() {
+        if (getDiagramId().isPresent()) {
```
Comment (jstastny-cz) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/factory/graph/DMNSetConnectionTargetNodeCommand.java:99]: again, please introduce variable.

**Review by jstastny-cz — CHANGES_REQUESTED** [review]
Please respond to the Sonar check issues introduced by this PR.

**Comment by jstastny-cz** [comment]
jenkins do fdb

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/factory/graph/DMNSetConnectionSourceNodeCommand.java (hunk 2 above)
Comment (danielzhe) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/factory/graph/DMNSetConnectionSourceNodeCommand.java:50]: The problem here is that I'm just returning something from the super class. I think that to fix that we would need to do a big refactor in other classes not involved in this PR.

Comment (jstastny-cz) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/factory/graph/DMNSetConnectionSourceNodeCommand.java:50]: I thought so, just wanted to clarify. IMO this should not block this PR.

**Comment by danielzhe** [comment]
Changes applied, @jstastny-cz! Thank you!

**Comment by danielzhe** [comment]
jenkins do fdb

**Comment by danielzhe** [comment]
@jstastny-cz Ready for your review. Thank you! =)

**Review by jstastny-cz — APPROVED** [review]
LGTM, verified using the FDB BC war.

**Pull request merged**